### PR TITLE
✨(backend) change course glimpse footer state

### PR DESCRIPTION
**Pull request by jamoqs**
This commit change text and remove datetime on glimpse footer.

## Purpose
Change text and remove datime on glimpse footer when state is ONGOING_OPEN 

## Proposal
This commit resolve the issue https://github.com/openfun/richie/issues/1970

- Changes was made on course models
- Existent tests was refactored.

### CODE REVIEW

File: src/richie/apps/courses/models/course.py [hunk 1]
```diff
@@ -110,12 +110,15 @@ def __init__(self, priority, date_time=None):
...
             date_time = None
+
         kwargs = {
             "priority": priority,
             "datetime": date_time,
             "call_to_action": self.STATE_CALLS_TO_ACTION[priority],
             "text": text,
         }
+        if priority == CourseState.ONGOING_OPEN:
+            kwargs["datetime"] = None
```
Comment (jbpenrath) [inline src/richie/apps/courses/models/course.py:121]: Instead you could edit the condition statement above, WDYT ?

```python
        if (
            priority in [CourseState.ONGOING_OPEN, CourseState.ARCHIVED_OPEN]
            and date_time.year == MAXYEAR
        ):
            text = _("forever open")
            date_time = None
        elif priority == CourseState.ONGOING_OPEN:
            date_time = None
```

File: src/frontend/js/types/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Nullable } from 'types/utils';
 
 export enum CourseStateTextEnum {
-  CLOSING_ON = 'closing on',
+  CLOSING_ON = 'open for enrollment',
```
Comment (jbpenrath) [inline src/frontend/js/types/index.ts:4]:
```suggestion
  ENROLLMENT_OPENED = 'open for enrollment',
```

File: src/richie/apps/courses/models/course.py (hunk 1 above)
Comment (jamoqs) [reply src/richie/apps/courses/models/course.py:121]: I think this is a better idea, most of the times i have tried to follow pattern of the code and same parts a saw some if's.
In this exactly point i take a time to think do or don't hahaha, sorry for that.
Thank you, for revision, tomorrow at morning i will made changes at commit of this PR.

File: src/frontend/js/types/index.ts (hunk 2 above)
Comment (jamoqs) [reply src/frontend/js/types/index.ts:4]: I have a doubt to, but you give me a confirmation, thanks. =)

**Comment by sampaccoud** [comment]
@jamoqs do you want me to finalize this PR? We would like to merge it.

**Comment by jamoqs** [comment]
> @jamoqs do you want me to finalize this PR? We would like to merge it.

It's finished, now.

**Pull request merged**